install_config/persistent_storage/pod_security_context.adoc

```diff
@@ -12,262 +12,728 @@ toc::[]
 ====
 
-== SupplementalGroups
+[TIP]
+====
+It is generally preferable to use group IDs (supplemental or
+link:#fsgroup[fsGroup]) to gain access to persistent storage versus using
+link:#user-id[user IDs].
+====
 
 Supplemental groups are regular Linux groups. When a process runs in Linux, it
-has a UID and one or more GIDs. Usually this is just the UID and GIDs of the
-user who started the process, but in the case of containers you have the option
-of setting these attributes for the container's main process.
+has a UID, a GID, and one or more supplemental groups. These attributes can be
+set for a container's main process. The `*supplementalGroups*` IDs are typically
+used for controlling access to shared storage, such as NFS and GlusterFS,
+whereas link:#fsgroup[fsGroup] is used for controlling access to block storage,
+such as Ceph RBD and iSCSI.
+
+The OpenShift shared storage plug-ins mount volumes such that the POSIX
+permissions on the mount match the permissions on the target storage. For
+example, if the target storage's owner ID is 1234 and its group ID is 5678, then
+the mount on the host node and in the container will have those same IDs.
+Therefore, the container's main process must match one or both of those IDs in
+order to access the volume.
 
-The `*SupplementalGroups*` option of the pod `*SecurityContext*` stanza enables
-you to specify a list of groups which will be added to the main process of every
-container in the pod. This is useful because it enables you to give your pods
-group based access to volumes.
+[[nfs-example]]
+For example, consider the following NFS export.
 
-For example, consider the following NFS export:
+On an OpenShift node:
 
+[NOTE]
+====
+`showmount` requires access to the ports used by `rpcbind` and `rpc.mount` on
+the NFS server)
 ====
-----
-showmount -e example.com
-Export list for example.com:
-/nfsexport (everyone)
 
-# ls -ld /nfsexport # on NFS server
-drwxrwx---. 2 root 1234 4096 Oct 30 15:27 /nfsexport
+====
 ----
+# showmount -e <nfs-server-ip-or-hostname>
+Export list for f21-nfs.vm:
+/opt/nfs  *
+----
+====
+
+On the NFS server:
+
 ====
+----
+# cat /etc/exports
+/opt/nfs *(rw,sync,no_root_squash)
+...
 
-The above export is only accessible by *root* and the group *1234*. So any pod
-which is not run as *root* and is not a member the group *1234* will not be able
-to access that export. It is often the case that the primary UID in the
-container cannot be controlled or predicted.
+# ls -lZ /opt/nfs -d
+drwxrws---. nobody 5555 unconfined_u:object_r:usr_t:s0   /opt/nfs
 
-To solve this issue, you can add the group *1234* to `*SupplementalGroups*` as
-in the following example:
+# id nobody
+uid=99(nobody) gid=99(nobody) groups=99(nobody)
+----
+====
 
+[NOTE]
 ====
+In the above, the owner is 99 (*nobody*), but the suggestions and examples in
+this topic apply to any non-root owner.
+====
+
+The *_/opt/nfs/_* export is accessible by UID *99* and the group *5555*. In
```

Comment:
ISTR directories should end in slash, but then there are several instances in the docs of them w/o.  (shrug)